packages/dev/src/main.ts

```diff
@@ -537,8 +547,12 @@ export class NetlifyDev {
     }
 
     if (this.#features.images) {
+      const remoteImages = [...(this.#config?.config?.images?.remote_images ?? []), ...this.#imageAllowedDomains]
+
       this.#imageHandler = new ImageHandler({
-        imagesConfig: this.#config?.config.images,
+        imagesConfig: {
+          remote_images: remoteImages,
```

Comment:
🤔 why was this snake_case? either it's just programmatic config for `ImageHandler`, in which case it should be camelCase, or it's intended to be the raw config from `netlify.toml`, in which case we shouldn't merge the dev package's `imageAllowedDomains` into it.

Reply:
(while we're at it, maybe `remoteImageAllowedPatterns` or something)

Reply:
Can you cut an issue so we can do this as a follow-up?